### PR TITLE
Fixes ACDC documents retrieval. LumiBased splitting at least

### LINKED ISSUE #5665: ACDC does not work for LumiBased splitting
During the last cmsweb-testbed validation, we noticed ACDC wfs using LumiBased splitting are not retrieving all the lumis that need to be recovered (or it's not retrieving all job failures from ACDC database).

### PR DESCRIPTION
**Pull request by amaltaro**
What a tough bug to track...
Should fix #5665 , since we were updating the same key (when > 1 failures in the same input file).
Tested interactively only, I still need to test the whole chain, so don't merge yet.

### CODE REVIEW

**Comment by amaltaro** [comment]
Question, is it possible that a single production/processing job process more than 1 file? If so, then I need to update this patch...

**Comment by ticoann** [comment]
I am not completely sure but code doesn't prevent having multiple input files. Also, there is merge jobs which take more then one input files. I guess we are not running ACDC's over merge jobs, do we?
When I see the ErrorHanlder there are a lot jobs using the same input file but there run lumi was different. I was wondering what kind of jobs were they.

**Comment by hufnagel** [comment]
Sure it is, all depends on what the job splitting does. The AlcaSkimming step for instance does this. It runs on unmerged files and is scheduled more like a merge job, but is internally a processing job.

**Comment by amaltaro** [comment]
@ticoann , yes we do run ACDCs over merge and I just noticed this patch does not work for that case. Working on it.

**Comment by amaltaro** [comment]
Hopefully the last update :-) Works interactively! I'm testing it in my private node now.

**Comment by amaltaro** [comment]
@ticoann I could not test it further because all my jobs are stuck..
If you agree, I would merge it for tomorrow's testbed tag and I'll handle it in high prio during test phase such that if it breaks something, we still have time to fix it :-)

File: src/python/WMCore/ACDC/DataCollectionService.py [hunk 1]
```diff
@@ -96,14 +96,13 @@ def _getFilesetInfo(self, collectionName, filesetName, user, group,
...
+        filesInfo = []
         for row in results["rows"]:
             files = row["doc"].get("files", False)
-            if (files):
-                filesetFiles.update(files)
-        
-        filesInfo = filesetFiles.values()
-        
+            if files:
+                filesInfo.append(files.values())
```
Comment (ticoann) [inline src/python/WMCore/ACDC/DataCollectionService.py]: Shouldn't this be filesInfo.extend(files.values()) instead of append?

**Comment by amaltaro** [comment]
The final result will be the same, but extend() looks cleaner. Patch updated.
 Good catch Dirk ;)

**Comment by ticoann** [comment]
Actually append and extend will return different format. In this case append will return list of list (of dict) but when use extend, it will return list (of dict). The second case is what you want, right?
To be clear, this patch fixes the case, when different job has the same input files, correct?

**Comment by amaltaro** [comment]
That't right! Extend will create a list of dict, which is what we need in the end.
And you're correct again, different jobs with the same input file won't overwrite its value anymore.

**Comment by ticoann** [comment]
Thanks Alan for the clarification. I am merging this.

**Pull request merged**